src/least_satisfying.rs

```diff
@@ -174,7 +174,11 @@ pub enum Satisfies {
 
 impl fmt::Display for Satisfies {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{:?}", self)
+        match self {
+            Self::Yes => write!(f, "Does not reproduce the regression"),
+            Self::No => write!(f, "Reproduces the regression"),
```

Comment:
I think this might be backwards.

`Yes` means it reproduces the regression (rustc failed to compile).
`No` means it did not reproduce (rustc compiled successfully).

However, the terms "regress" and "not regress" may not be completely clear. For example, you might be looking for when something was *fixed*, such as when using `--regress=no-error`, which has an opposite meaning ("no" means "fails to compile", "yes" means "compiles successfully").

I would be fine with merging this with the text reversed, since it is a little clearer than "yes" and "no", but I suspect there is still room for confusion here. I think longer term it would be better to have customizable terms, and then things like `--regress=no-error` could change them to "compiles successfully" and "failed to compile", and offer the ability to customize the message like `git bisect` does (maybe `--term-old=good --term-new=bad`).

Reply:
thanks @ehuss exactly the feedback I needed 

Ok I've fixed the "logical" error.

Now, for a better wording. The Yes/No in the end means whether the toolchain at hand did produce the expected result using the sample code provided, correct? "Yes" (it did) or "No" (it didn't).

I couldn't think of a single sentence to squeeze all "success" cases and all "failure" cases into a single sentence. I also think that the message should be contextual to the `--regress` parameter used.

Unless we use some vague wording like "_requested condition matched_" or "_not matched_".

wdyt ?

Reply:
> Unless we use some vague wording like "requested condition matched" or "not matched".

That actually sounds a little better to me. I do worry that specifically saying `Reproduces the regression` could be confusing if you are searching for when the regression was fixed (since it will be backwards).

I opened #316 for being able to customize it. I can work on that sometime later unless you are interested.